numpy/core/memmap.py

```diff
@@ -236,11 +236,10 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
                 raise ValueError("Size of available data is not a "
                         "multiple of the data-type size.")
             size = bytes // _dbytes
-            shape = (size,)
         else:
             if not isinstance(shape, tuple):
                 shape = (shape,)
-            size = 1
+            size = np.intp(1) # Causes bytes below to be computed as the integer type the OS uses for sizes - can prevent overflow
```

Comment:
Even this is not true - `bytes` below is always a `long`. No need to explain what `np.intp` means here either. Perhaps instead  `# avoid default choice of np.int_, which might overflow`

nit: PEP8 wants two spaces before the `#`